WarshipGame/Assets/Scripts/Gameplay/Grid/GraphSearch.cs

```diff
@@ -0,0 +1,100 @@
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+
+/// <summary>
+/// Calculates the range of the character, the cost of the path and the possibility of the range and path
+/// </summary>
+public class GraphSearch
+{
+    //Breadth First Search Result*
+    public static BFSResult BFSGetRange(HexGrid hexGrid, Vector2Int startPoint, int movementPoints)
+    {
+        Dictionary<Vector2Int, Vector2Int?> visitedHexes = new Dictionary<Vector2Int, Vector2Int?>();
+        Dictionary<Vector2Int, int> rangeCost = new Dictionary<Vector2Int, int>();
+        Queue<Vector2Int> hexesToVisitQueue = new Queue<Vector2Int>();
```

Comment:
`new Dictionary<>()` and `new Queue<>()` is redundant and can be changed into `new()`